File: packages/client/ios/RNOta/Rollback/RollbackResult.swift
import Foundation

public struct RollbackResult: Equatable, Sendable {
  public var performed: Bool
  public var decision: RollbackDecision
  public var reason: RollbackReason
  public var fromSlotId: String?
  public var toSlotId: String?
  public var activeSlotAfter: String?
  public var pendingCleared: Bool
}

File: packages/client/android/src/main/java/com/rnota/rollback/RollbackResult.kt
package com.rnota.rollback

/**
 * Outcome of a rollback API call.
 *
 * @property toSlotId null means embedded bundle.
 */
data class RollbackResult(
  val performed: Boolean,
  val decision: RollbackDecision,
  val reason: RollbackReason,
  val fromSlotId: String?,
  val toSlotId: String?,
  val activeSlotAfter: String?,
  val pendingCleared: Boolean,
)
